### DISCUSSION TITLE
What determines what the governor builds? Any way to customize it?

### DISCUSSION THREAD

**Discussion by makotech222** [induktio/thinker#77] [General]
I generally play with all my cities on Governor, usually Build/Discover, but generally am unhappy with the units they produce and which buildings they choose.

Unit-wise, they seem to choose alien units alot (mind worms mostly) instead of my custom designed units that i prefer. For buildings, they often skip good buildings like research hospital, and choose stuff like punishment sphere which has significant repercussions .

It would be great if there were some way to customize what each governor AI chooses to build. Otherwise, the thinker mod has been absolutely great and I love it!

**Comment by induktio** [comment]
Governor priorities and build orders are part of the mod code that is compiled so there isn't currently any more specific way customize it other than adjusting governor settings and using the build priorities. PSI combat units are chosen more often if the faction has any PSI combat bonuses or secret projects, base facilities increasing the lifecycle or high Planet rating. Likewise Punishment Spheres are chosen more often on captured bases with high inefficiency and drone counts and no Rec Commons or Hologram Theatres present.

The current governor options menu has only limited choices which do not influence these directly. Not sure how easy it is to patch but it would be possible add somewhere extra settings for player bases so that some build choices could be skipped. At least PSI combat units, Punishment Spheres, or crawler units might be useful as options, maybe there are others? Going even further it might be possible to move the governor build order to its own config file but not sure if it's that necessary.

**Reply by makotech222** [reply]
Any enhancement would be appreciated by me. I just hate seeing my cities waste a turn making stuff I don't want, and I usually have too many cities to manually control in an enjoyable way.

**Reply by makotech222** [reply]
Relatedly, what are the criteria for the governor to Hurry production? I have tons of Energy available, but i don't ever notice governor using it.

**Reply by induktio** [reply]
The function that decides it is in build.cpp > mod_base_hurry and it is quite long. Both the hurry option and governor is active needs to be enabled from governor options. The governor keeps some credits in reserve so you should have a couple of hundred credits available.
